lib/hooks/useCustomEvent.tsx

```diff
@@ -0,0 +1,55 @@
+import { useRef } from "react";
+
+/**
+ *  @TODO: Define unique payload types for each CustomEvent and add
+ *         to CustomEventDetails instead of using generic object
+ */
+export type CustomEventDetails = object | undefined;
+
+export const useCustomEvent = () => {
+  // Attach listeners to a documentRef instead of document directly
+  // https://www.preciousorigho.com/articles/a-better-way-to-create-event-listeners-in-react
+  const documentRef = useRef<Document | null>(null);
+
+  if (typeof window !== "undefined") {
+    documentRef.current = window.document;
+  }
+
+  const Event = {
+    /**
+     * Fire an event, pass an optional payload
+     * @param eventName string
+     * @param data CustomEventDetails
+     */
+    fire: (eventName: string, data: CustomEventDetails = undefined) => {
+      const event = new CustomEvent(eventName, { detail: data });
```

Comment:
Should we just call this data (not detail)

Reply:
detail is the name from the underlying CustomEvent class ... so renaming our custom data prop to match for consistency